### ISSUE TITLE
Incorrect nullability of nested struct field

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#34700]
No warning should be reported for `t.Item1.F.ToString()`:
```C#
struct S
{
    internal object? F;
}

class Program
{
    static void Main()
    {
        S s = new S() { F = 1 };
        var t = (s, s = new S());
        t.Item1.F.ToString();
        t.Item2.F.ToString(); // warning
    }
}
```
Result:
```
(12,9): warning CS8602: Dereference of a possibly null reference.
(13,9): warning CS8602: Dereference of a possibly null reference.
```
